src/invocation/InvocationService.ts

```diff
@@ -213,8 +224,35 @@ export class Invocation {
     }
 
     complete(clientMessage: ClientMessage): void {
-        this.deferred.resolve(clientMessage);
-        this.invocationService.deregisterInvocation(this.request.getCorrelationId());
+        /**
+         * The following is a part of eager deserialization that is needed for compact serialization. In
```

Comment:
Well it is not the main reason, however we expect some performance benefits of getting rid of another promise. I remember I measured it and there was an improvement. 

I think the main reason was to move toObject callbacks to a single place so that we don't have to write try/catch blocks everywhere. If we did not do it, then every proxy method would be have to handle this.